### PR TITLE
pipe_tasks/DM-8221: add aperture corrections and schema checks to testCoadds

### PR DESCRIPTION
**Pull request by TallJimbo**
This diff is relative to DM-8210.  I'll rebase after that is merged.

### CODE REVIEW

File: python/lsst/pipe/tasks/mocks/mockObservation.py [hunk 1]
```diff
@@ -214,3 +220,27 @@ def buildPsf(self, detector):
             spatialFuncList
         )
         return lsst.meas.algorithms.KernelPsf(kernel)
+
+    def buildApCorrMap(self, detector):
+        """Build an ApCorrMap with random linearly-varying fields for all
+        flux fields registered for aperture correction.
+
+        The given flux names are used only as strings; there is no connection
```
Comment (r-owen) [inline python/lsst/pipe/tasks/mocks/mockObservation.py]: Please document where the flux names come from.

Comment (TallJimbo) [inline python/lsst/pipe/tasks/mocks/mockObservation.py]: Done.  The answer - from the aperture correction registry - was actually in the previous sentence, but this sentence (which was a relic of a version that took the flux names as arguments) confused that.

**Comment by TallJimbo** [comment]
Switched to `import numpy as np`.

**Comment by TallJimbo** [comment]
Closing after manual merge; PR base is not master.

**Pull request closed without merging**